vars/gradleWrapper.groovy

```diff
@@ -1,18 +1,32 @@
 #!/usr/bin/env groovy
 
-def cleanupCommand(String command) {
+/**
+ * @param command
+ * @return A formatted command for the gradle process. If a log level was provided by the
+ * environment it will be appended to the command.
+ */
+def formatCommand(String command) {
    regex = /-P[\w\d.]+=\s/
-   command.replaceAll(regex, '')
+   command = command.replaceAll(regex, '')
+
+    if (params.LOG_LEVEL){
+        command += " --${params.LOG_LEVEL}"
+    }
+    else if (env.LOG_LEVEL) {
+        command += " --${env.LOG_LEVEL}"
+    }
+    return command
 }
```

Comment:
Yes cool. Only thing still is to check if the command wants to set a log level explicitly. Which should override the other selection.

Reply:
Hmm, should the order of precedence not be?

1. Params
2. Config
3. What's provided by command